src/app-layer-parser.c

```diff
@@ -919,6 +918,23 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint32_t tofree_nb = 0;
+    if (total_txs > UINT32_MAX) {
+        SCFree(txl->tx_id_free_list);
+        txl->tx_id_free_list = NULL;
+        txl->max_tx = 0;
+    } else if (txl->max_tx < total_txs) {
+        uint64_t *tmp = SCRealloc(txl->tx_id_free_list, total_txs * sizeof(uint64_t));
```

Comment:
can we just double the size every time we need more space? No need to realloc for every additional tx in a flow then. Could just increase to the next power of 2 that fits the number of txs we need.

Reply:
ok, increasing by a step of 256